dss/storage/index_document.py

```diff
@@ -15,26 +15,20 @@
 from dss import Config, DeploymentStage, ESIndexType, ESDocType
 from dss import Replica
 from dss.hcablobstore import BundleMetadata, BundleFileMetadata
-from dss.storage.bundles import BUNDLE_PREFIX, DSS_OBJECT_NAME_REGEX, DSS_BUNDLE_KEY_REGEX
-from dss.storage.bundles import bundle_key, bundle_key_to_bundle_fqid, bundle_fqid_to_uuid_version, format_bundle_fqid
+from dss.storage.bundles import ObjectIdentifier, BundleFQID, TombstoneID
 from dss.storage.index import Index
 from dss.util import create_blob_key
 from dss.util.es import ElasticsearchClient
 
 
 class IndexDocument(dict):
 
-    def __init__(self, replica: Replica, bundle_uuid: str, bundle_version: typing.Optional[str],
+    def __init__(self, replica: Replica, identifier: typing.Union[BundleFQID, TombstoneID],
                  logger: logging.Logger) -> None:
         super().__init__()
         self.logger = logger
         self.replica = replica
-        self.bundle_uuid = bundle_uuid
-        self.bundle_version = bundle_version
-
-    @property
-    def bundle_fqid(self):
-        return format_bundle_fqid(self.bundle_uuid, self.bundle_version)
+        self.identifier = identifier
```

Comment:
For the sake of brevity and recognizability, I would prefer this to be called fqid and rename ObjectIdentifier to be called FQID. I would gloss over the fact that a version-unspecific TombstoneID is not fully-qualified.